### PR TITLE
Diskcache

### PR DESCRIPTION
**Pull request by tilschaef**
Replaced current caching approach with [diskcache](http://www.grantjenks.com/docs/diskcache/index.html)

- unified caching approach
- short and long term storage with SQL backend
- thread safety

### CODE REVIEW

**Review by siebrenf — COMMENTED** [review]
I'm fixing the xenbase-related tests. If you can fix the linting and check the self argument we're good to go!

File: genomepy/annotation/mygene.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def _map_genes(
     # remove version numbers from gene IDs
     split_id = df["name"].str.split(r"\.", expand=True)[0]
     df = df.assign(split_id=split_id.values)
-    genes = set(split_id)
+    genes = sorted(set(split_id))
```
[the remark this answers is not in the document]
Comment (tilschaef) [reply genomepy/annotation/mygene.py:61]: Thanks :). I will have a look at the linting and see if I can ignore self somehow.

**Comment by siebrenf** [comment]
We ran out of test credits. I've asked Travis for more, and will try to restart the build in a bit

**Comment by siebrenf** [comment]
Fixed now. Seems Travis doesn't store requests while we're out of credits, so the tests will start when you add a new commit!

**Comment by tilschaef** [comment]
The PR seems to consistently fail on one of the CI builds, usually Linux, for `tests/test_14_functions.py::test_head_annotations`. On macOS, all tests pass without problems. I have a suspicion that it might be related to retrieving the cached annotation/provider.

File: genomepy/caching.py [hunk 2]
```diff
@@ -2,21 +2,23 @@
...
+from diskcache import Cache
 
 from genomepy.__about__ import __version__
+from genomepy.config import config
+
+# Cache expiration times (in seconds)
+cache_exp_short = 3.6e3
+cache_exp_long = 8.64e4
+# Set max size for cache (in bytes) from config
+cache_size_limit = config.get("cache_size_limit")
```
Comment (siebrenf) [inline genomepy/caching.py:14]: should there be a default value here (`600000000`)?

**Review by siebrenf — COMMENTED** [review]
tests/data/small_genome.fa.gz changed in size. Fine by me if tests still work.

**Review by siebrenf — APPROVED** [review]
wohoo! 🚀

**Pull request merged**